src/Nest/CommonAbstractions/ConnectionSettings/ConnectionSettingsBase.cs

```diff
@@ -63,17 +68,36 @@ public abstract class ConnectionSettingsBase<TConnectionSettings> : ConnectionCo
 		private readonly FluentDictionary<MemberInfo, IPropertyMapping> _propertyMappings = new FluentDictionary<MemberInfo, IPropertyMapping>();
 		FluentDictionary<MemberInfo, IPropertyMapping> IConnectionSettingsValues.PropertyMappings => _propertyMappings;
 
-		protected ConnectionSettingsBase(IConnectionPool connectionPool, IConnection connection, Func<TConnectionSettings, IElasticsearchSerializer> serializerFactory)
-			: base(connectionPool, connection, serializerFactory)
+		private readonly ISerializerFactory _serializerFactory;
+		ISerializerFactory IConnectionSettingsValues.SerializerFactory { get; }
+
+		protected ConnectionSettingsBase(
+			IConnectionPool connectionPool,
+			IConnection connection,
+			ISerializerFactory serializerFactory,
+			Func<TConnectionSettings, IElasticsearchSerializer> serializerFactoryFunc
+			)
+			: base(connectionPool, connection, serializerFactoryFunc)
 		{
 			this._defaultTypeNameInferrer = (t => t.Name.ToLowerInvariant());
 			this._defaultFieldNameInferrer = (p => p.ToCamelCase());
 			this._defaultIndices = new FluentDictionary<Type, string>();
 			this._defaultTypeNames = new FluentDictionary<Type, string>();
+			this._serializerFactory = serializerFactory ?? new SerializerFactory();
 
 			this._inferrer = new Inferrer(this);
 		}
 
+		protected ConnectionSettingsBase(
+			IConnectionPool connectionPool,
+			IConnection connection,
+			Func<TConnectionSettings, IElasticsearchSerializer> serializerFactoryFunc
+			)
+			: this(connectionPool, connection, null, serializerFactoryFunc) { }
+
+		IElasticsearchSerializer IConnectionSettingsValues.CreateStatefulSerializer(JsonConverter converter) =>
```

Comment:
For parity reasons, I think we should rename this to just `StatefulSerializer` (e.g.  `settings.Serializer` and `settings.StatefulSerializer`).